### DISCUSSION TITLE
Cannot close gripper using force control due to md5sum mismatch

### DISCUSSION THREAD

**Discussion by Keiichi-Ono** [HumanSupportRobot/QandA#99] [General]
I used apply_force(10) in ihsrb to close the gripper, but an error occurred.I think the cause of the error is that the value of md5sum is not the value requested by the Client.

How can I make the md5sum value match and  the gripper closes?
I would appreciate it if you could provide me with a solution.
 

Please see the warning below.

> <hsrb>~$ ihsrb
> [WARN] [1712293076.537424]: Could not process inbound connection: Client [/hsrb/robot_hardware] wants topic [/hsrb/gripper_controller/grasp/goal] to have datatype/md5sum [tmc_control_msgs/GripperApplyEffortActionGoal/43355ef33a9f9ba5d26e50a3cc6e60bd], but our version has [tmc_control_msgs/GripperApplyEffortActionGoal/09d77e47aded4b89a3aa6a2d855822a3] Dropping connection.{'callerid': '/hsrb/robot_hardware', 'md5sum': '43355ef33a9f9ba5d26e50a3cc6e60bd', 'tcp_nodelay': '0', 'topic': '/hsrb/gripper_controller/grasp/goal', 'type': 'tmc_control_msgs/GripperApplyEffortActionGoal'}
> [WARN] [1712293076.719584]: Could not process inbound connection: Client [/hsrb/robot_hardware] wants topic [/hsrb/gripper_controller/apply_force/goal] to have datatype/md5sum [tmc_control_msgs/GripperApplyEffortActionGoal/43355ef33a9f9ba5d26e50a3cc6e60bd], but our version has [tmc_control_msgs/GripperApplyEffortActionGoal/09d77e47aded4b89a3aa6a2d855822a3] Dropping connection.{'callerid': '/hsrb/robot_hardware', 'md5sum': '43355ef33a9f9ba5d26e50a3cc6e60bd', 'tcp_nodelay': '0', 'topic': '/hsrb/gripper_controller/apply_force/goal', 'type': 'tmc_control_msgs/GripperApplyEffortActionGoal'}
> 

**

> <hsrb>~$ rosmsg md5 tmc_control_msgs/GripperApplyEffortActionGoal 
> 09d77e47aded4b89a3aa6a2d855822a3

**

**Comment by yoshimi-iyoda** [comment]
It appears that the discrepancy in software versions between the client and the server could be the root cause. I would recommend upgrading to the latest software version to ensure compatibility. If you only intend to use it temporarily, please log in to the robot as indicated below and execute it in the Docker environment.

```
$ ssh ....
$ docker exec -it docker.hsrb.roscore.service bash
$ source /opt/ros/noetic/setup.bash
$ ihsrb
```

**Reply by Keiichi-Ono** [reply]
Thanks for the reply.

I ran gripper.applyforce under Docker environment and it worked without any errors!
In addition, I updated the version of the software on the HSR side, and now it can be run even without Docker environment.

Thanks to your advice, I was able to get it to work.
